packages/expo-contacts/android/src/main/java/expo/modules/contacts/models/BaseModel.java

```diff
@@ -12,173 +12,191 @@
+
+  protected void mapValue(Map<String, Object> readableMap, String key, String alias) {
+    if (readableMap.containsKey(key)) {
+      Object value = readableMap.get(key);
+      if (value instanceof Boolean) {
+        map.putBoolean(alias == null ? key : alias, (Boolean) readableMap.get(key));
+      } else {
+        map.putString(alias == null ? key : alias, (String) readableMap.get(key));
+      }
+    }
+  }
+
+  public void fromCursor(Cursor cursor) {
+    putString(cursor, getIdAlias(), EXColumns.ID);
+    map.putString(getLabelAlias(), getLabelFromCursor(cursor));
+    putString(cursor, getDataAlias(), EXColumns.DATA);
+    putString(cursor, EXColumns.LABEL, EXColumns.LABEL);
+    putString(cursor, getTypeAlias(), EXColumns.TYPE);
+    putInt(cursor, getIsPrimaryAlias(), EXColumns.IS_PRIMARY);
+  }
+
+  public final ContentProviderOperation getInsertOperation() {
+    return getInsertOperation(null);
+  }
+
+  public ContentProviderOperation getInsertOperation(String rawId) {
+    ContentProviderOperation.Builder op = ContentProviderOperation.newInsert(ContactsContract.Data.CONTENT_URI);
+    if (rawId == null) {
+      op.withValueBackReference(ContactsContract.Data.RAW_CONTACT_ID, 0);
+    } else {
+      op.withValue(ContactsContract.Data.RAW_CONTACT_ID, rawId);
+    }
+
+    return op.withValue(EXColumns.MIMETYPE, getContentType())
+      .withValue(EXColumns.TYPE, mapStringToType(getLabel()))
+      .withValue(EXColumns.DATA, getData())
+      .withValue(EXColumns.ID, getId())
+      .build();
+  }
+
+  public ContentProviderOperation getDeleteOperation(String rawId) {
+    return ContentProviderOperation.newDelete(ContactsContract.Data.CONTENT_URI)
+      .withSelection(
+        String.format("%s=? AND %s=?", ContactsContract.Data.MIMETYPE, ContactsContract.Data.RAW_CONTACT_ID),
+        new String[]{getContentType(), rawId})
+      .build();
+  }
+
+  public String getId() {
+    return getString(getIdAlias());
+  }
+
+  public String getLabel() {
+    return getString(getLabelAlias());
+  }
+
+  public String getData() {
+    return getString(getDataAlias());
+  }
+
+  public String getType() {
+    return getString(getTypeAlias());
+  }
+
+  public int getIsPrimary() {
+    if (map.containsKey(getIsPrimaryAlias()))
+      return map.getInt(getIsPrimaryAlias());
+    return 0;
+  }
+
+  public String getString(String key) {
+    if (map.containsKey(key))
+      return map.getString(key);
+    return null;
+  }
+
+  public void fromMap(Map<String, Object> readableMap) {
+    for (String key : readableMap.keySet()) {
+      mapValue(readableMap, key);
+    }
+  }
+
+  public Bundle getMap() {
+    return map;
+  }
+
+  protected String getLabelFromCursor(Cursor cursor) {
+    switch (cursor.getInt(cursor.getColumnIndex(EXColumns.TYPE))) {
+      case EXColumns.TYPE_CUSTOM:
+        final String label = cursor.getString(cursor.getColumnIndex(EXColumns.LABEL));
+        return label != null ? label : "unknown";
+      default:
```

Comment:
Are there any changes apart from formatting 😅 ?

Reply:
In this file nope. 
Please, check `ArrayList<ContentProviderOperation> toUpdateOperationList()` method and places where it is use ;)